src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,195 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, {
+    Children,
+    cloneElement,
+    FC,
+    isValidElement,
+    ReactChild,
+    ReactNode,
+    useEffect,
+    useRef,
+    useState,
+} from "react";
+import { TabItemProps } from "@components/Tabs/TabItem";
+import { merge } from "@utilities/merge";
+import { IconMore } from "@foundation/Icon";
+import { Badge } from "@components/Badge";
+import { motion } from "framer-motion";
+import { Button, ButtonSize, ButtonStyle } from "@components/Button";
+import { Flyout } from "@components/Flyout";
+import { MenuItem, MenuItemStyle } from "@components/MenuItem";
+
+export type Content = {
+    tabItem: TabItemProps;
+    content: ReactChild;
+};
+
+export enum TabsPaddingX {
+    Small = "Small",
+    Medium = "Medium",
+    Large = "Large",
+}
+
+export enum TabSize {
+    Small = "Small",
+    Large = "Large",
+}
+
+export type TabsProps = {
+    paddingX: TabsPaddingX;
+    size: TabSize;
+    activeItemId: string;
+    children: ReactNode;
+    onChange: (tabId: string) => void;
+};
+
+const paddingMap: Record<TabsPaddingX, string> = {
+    Small: "tw-pl-s",
+    Medium: "tw-pl-m",
+    Large: "tw-pl-l",
+};
+
+export const Tabs: FC<TabsProps> = ({ paddingX, size, activeItemId, children, onChange }) => {
+    const [isOverflowing, setIsOverflowing] = useState(false);
+    const [isMenuOpened, setIsMenuOpened] = useState(false);
+
+    const tabs: TabItemProps[] =
+        Children.map(children, (child) => {
+            if (!isValidElement<TabItemProps>(child)) {
+                return null;
+            }
+            return child?.props;
+        }) ?? [];
+
+    const [tabIndexLimit, setTabIndexLimit] = useState(tabs.length);
+
+    const checkIfOverflowing = () => {
+        const tabNav = tabNavRef.current;
+        setIsOverflowing((tabNav && tabNav.scrollWidth > tabNav.clientWidth) || false);
```

Comment:
```suggestion
        setIsOverflowing((tabNav && tabNav.scrollWidth > tabNav.clientWidth) ?? false);
```
Prefer `??` when working with boolean or empty string